Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java

```diff
@@ -0,0 +1,133 @@
+     * generated data for project blog page by fetching project blog entries
+     * from <code>tbl_projectblogentry</code> and checking user privileges.
+     * 
+     * @param projectUnixName
+     *            parameter passed from url as {projectUnixName}
+     * @param model
+     *            model object
+     * @param project
+     *            project instance obtained using @InjectProject annotation
+     * @param principal
+     *            principal object which is required to fetch information about
+     *            logged in user.
+     * @return project blog page containing list of all project blogs from given
+     *         project
+     * @throws QuadrigaStorageException
+     * @throws NoSuchRoleException
+     */
+    @RequestMapping(value = "sites/{ProjectUnixName}/projectblog", method = RequestMethod.GET)
+    public String projectBlog(@PathVariable("ProjectUnixName") String projectUnixName, Model model,
+            @InjectProject(unixNameParameter = "ProjectUnixName") IProject project, Principal principal)
+            throws QuadrigaStorageException, NoSuchRoleException {
+
+        // Fetch blog entries for a project identified by project unix name
+        String projectId = project.getProjectId();
+        Integer count = getBlogEntryCount();
+        List<IProjectBlogEntry> projectBlogEntryList = projectBlogEntryManager.getProjectBlogEntryList(projectId,
+                count);
+
+        // Add the critical data to model object
+        model.addAttribute("projectBlogEntryList", projectBlogEntryList);
+        model.addAttribute("project", project);
+
+        // check the roles of logged in user to decide whether to show add
+        // project link or not. Only project admins and owners are allowed to
+        // add project blogs for a project under consideration
+        if (principal != null) {
+            String userName = principal.getName();
+            if (projectSecurity.isCollaborator(userName, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN,
+                    project.getProjectId())) {
+                model.addAttribute("isProjectAdmin", true);
+            } else {
+                model.addAttribute("isProjectAdmin", false);
+            }
+
+            if (projectSecurity.isProjectOwner(userName, project.getProjectId())) {
+                model.addAttribute("isProjectOWner", true);
+            } else {
+                model.addAttribute("isProjectOWner", false);
+            }
+        } else {
+            model.addAttribute("isProjectAdmin", false);
+            model.addAttribute("isProjectOWner", false);
+        }
+
+        return "sites/projectblog";
+    }
+
+    /**
+     * reads the number of blog entries to be displayed from user properties
+     * file using {@linkplain Environment} object
+     * 
+     * @return number of blog entries to be displayed specified as
+     *         <code>projectblogentry.list.count<code> property.
+     */
+    private Integer getBlogEntryCount() {
```

Comment:
Well, ideally, the the blog would work something like that:
1. there is a default number of blog entries shown on the page. Default should be retrieved in manager class), so that there is no code duplication, if another class uses the same method (to retrieve all blog entries)
2. the blog page might let the user specify that they want to see more or less blog entries on the page (so the controller can call a method in the manager, that let's them specify how many blog entries they have). Basically, controllers control what's coming from the user side, manager control what the user can't influence (e.g. default values).
3.